mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/vault/attachements/AttachmentsPresenter.kt

```diff
@@ -18,52 +20,52 @@ import rs.readahead.washington.mobile.media.MediaFileHandler.walkAllFilesWithDir
-                    disposables.add(MyApplication.rxVault.list(vaultFile, filterType, sort, null)
-                        .subscribeOn(Schedulers.io())
-                        .doOnSubscribe { view?.onGetFilesStart() }
-                        .observeOn(AndroidSchedulers.mainThread())
-                        .doFinally { view?.onGetFilesEnd() }
-                        .subscribe(
-                            { vaultFiles: List<VaultFile?> ->
-                                view?.onGetFilesSuccess(
-                                    vaultFiles
-                                )
-                            }
-                        ) { throwable: Throwable? ->
-                            FirebaseCrashlytics.getInstance().recordException(throwable!!)
-                            view?.onGetFilesError(throwable)
-                        })
-                }
-            ) { throwable: Throwable? ->
-                FirebaseCrashlytics.getInstance().recordException(throwable!!)
-                view?.onGetRootIdError(throwable)
-            }.dispose()
+        MyApplication.rxVault.get(parent).subscribe({ vaultFile: VaultFile? ->
+            disposables.add(MyApplication.rxVault.list(vaultFile, filterType, sort, null)
+                .subscribeOn(Schedulers.io()).doOnSubscribe { view?.onGetFilesStart() }
+                .observeOn(AndroidSchedulers.mainThread()).doFinally { view?.onGetFilesEnd() }
+                .subscribe({ vaultFiles: List<VaultFile?> ->
+                    view?.onGetFilesSuccess(
+                        vaultFiles
+                    )
+                }) { throwable: Throwable? ->
+                    FirebaseCrashlytics.getInstance().recordException(throwable!!)
+                    view?.onGetFilesError(throwable)
+                })
+        }) { throwable: Throwable? ->
+            FirebaseCrashlytics.getInstance().recordException(throwable!!)
+            view?.onGetRootIdError(throwable)
+        }.dispose()
     }
 
+    @SuppressLint("CheckResult")
     override fun importVaultFiles(uris: List<Uri?>, parentId: String?, deleteOriginal: Boolean) {
-        disposables.add(Observable.fromCallable {
-            MediaFileHandler.importVaultFilesUris(
-                view?.getContext(), uris, parentId
-            )
-        }
-            .subscribeOn(Schedulers.computation())
+        var counter = 1
+        disposables.add(Flowable.fromIterable<Uri>(uris).flatMap { uri ->
+            MediaFileHandler.importVaultFileUri(view?.getContext(), uri, parentId).toFlowable() }
+            .doOnComplete { view?.onImportEnded() }
             .doOnSubscribe { view?.onImportStarted() }
             .observeOn(AndroidSchedulers.mainThread())
             .doFinally { view?.onImportEnded() }
-            .subscribe(
+            .subscribeOn(Schedulers.computation()).observeOn(AndroidSchedulers.mainThread()).doOnNext {
+            if (counter == 1) {
+                counterData.postValue(counter)
```

Comment:
This is not correct as the consumer is not yet created, you need to call the callback first.